test/functional/core/tailable_cursor.test.js

```diff
@@ -1,42 +1,61 @@
 'use strict';
 const { expect } = require('chai');
-const { setupDatabase, withClientV2 } = require('../shared');
+const { runLater } = require('../../tools/utils');
 
 describe('Tailable cursor tests', function () {
-  before(function () {
-    return setupDatabase(this.configuration);
-  });
-
-  it('should correctly perform awaitData', {
-    metadata: { requires: { mongodb: '>=3.2' } },
-    test: withClientV2((client, done) => {
-      const db = client.db();
-      db.collection('cursor_tailable').drop(() => {
-        db.createCollection('cursor_tailable', { capped: true, size: 10000 }, (err, coll) => {
-          expect(err).to.not.exist;
+  describe('awaitData', () => {
+    let client;
 
-          coll.insertOne({ a: 1 }, (err, res) => {
-            expect(err).to.not.exist;
-            expect(res).property('insertedId').to.exist;
+    beforeEach(async function () {
+      client = this.configuration.newClient();
+      await client.connect();
+    });
 
-            const cursor = coll.find({}, { batchSize: 2, tailable: true, awaitData: true });
-            cursor.next((err, doc) => {
-              expect(err).to.not.exist;
-              expect(doc).to.exist;
+    afterEach(async () => {
+      await client.close();
+    });
 
-              const s = new Date();
-              cursor.next(() => {
-                const e = new Date();
-                expect(e.getTime() - s.getTime()).to.be.at.least(300);
+    it(
+      'should block waiting for new data to arrive when the cursor reaches the end of the capped collection',
+      {
+        metadata: { requires: { mongodb: '>=3.2' } },
+        async test() {
+          const db = client.db('cursor_tailable');
 
-                done();
-              });
+          try {
+            await db.collection('cursor_tailable').drop();
+            // eslint-disable-next-line no-empty
+          } catch (_) {}
 
-              setTimeout(() => cursor.close(), 300);
-            });
+          const collection = await db.createCollection('cursor_tailable', {
+            capped: true,
+            size: 10000
           });
-        });
-      });
-    })
+
+          const res = await collection.insertOne({ a: 1 });
+          expect(res).property('insertedId').to.exist;
+
+          const cursor = collection.find({}, { batchSize: 2, tailable: true, awaitData: true });
+          const doc = await cursor.next();
+          expect(doc).to.exist;
+
+          // After 300ms make an insert
+          const later = runLater(async () => {
+            const res = await collection.insertOne({ a: 1 });
+            expect(res).property('insertedId').to.exist;
+          }, 300);
+
+          const start = new Date();
+          await cursor.next();
```

Comment:
do we care what this returns? and that it specifically returns the newly inserted thing? (would need to update the actual object in L44 to differentiate)